### ISSUE TITLE
Support loading & saving ontologies to SPARQL stores  

### ISSUE THREAD

**Issue by sesuncedu** [owlcs/owlapi#600]
Also, support separate store for a-box. 

Can use openclipsdf4j.  

Might be interesting with stardoggy.

**Comment by ansell** [comment]
I agree that the current workflow for this with owlapi-rio is not optimal and needs some snappy helper methods to make it more seamless.

**Comment by sesuncedu** [comment]
Hmm. Is there a Handler adapter that can queue Statements, and feed them into a stream, ending the stream when end  document is called?

**Comment by ansell** [comment]
RioMemoryTripleSource, which accepts CloseableIteration<Statement>, which is provided by all of the RepositoryConnection and similar APIs:

https://github.com/owlcs/owlapi/blob/version5/rio/src/main/java/org/semanticweb/owlapi/rio/RioMemoryTripleSource.java

is specifically handled by RioParserImpl in a streaming fashion:

https://github.com/owlcs/owlapi/blob/version5/rio/src/main/java/org/semanticweb/owlapi/rio/RioParserImpl.java#L109

**Comment by ignazio1977** [comment]
That's for reading from a RDF store - for saving, the current interfaces are not great.

I'm working on a refactor in version 6, where document source and target will handle the output; storers able to save to traditional stream/writers and storers able to save to other stuff will be easier to mix.

**Comment by ansell** [comment]
Yes, I currently avoid the save pipeline and use RioRenderer:

https://github.com/owlcs/owlapi/blob/version5/rio/src/main/java/org/semanticweb/owlapi/rio/RioRenderer.java

**Comment by ansell** [comment]
It appears as though at some stage I made RioStorer work for streaming, but I haven't converted the small pieces of code I use it with yet:

https://github.com/owlcs/owlapi/blob/version5/rio/src/main/java/org/semanticweb/owlapi/rio/RioStorer.java

**Comment by ansell** [comment]
Or someone else made it work ;)
